qiskit/transpiler/passes/utils/merge_adjacent_barriers.py

```diff
@@ -57,6 +57,7 @@ class MergeAdjacentBarriers(TransformationPass):
 
     def run(self, dag):
         """Run the MergeAdjacentBarriers pass on `dag`."""
+        indices = {qubit: index for index, qubit in enumerate(dag.qubits)}
```

Comment:
Lol, it strikes again. This also was a likely source of a performance regression in how I implemented #9569 . We really need to implement #9389 and cache this in the dagcircuit.